File: frontend/src/pwa/qualification/formatting.js
/** P003.5 — Human-readable qualification receipt formatter. */
export function formatQualificationReceipt(receipt) {
  const lines = [
    "P003.5 INSTALLATION AND OFFLINE QUALIFICATION",
    "=".repeat(72),
    `Qualification: ${receipt.qualificationId}`,
    `Status: ${receipt.status}`,
    `Application version: ${receipt.applicationVersion}`,
    `Cache version: ${receipt.cacheVersion}`,
    `Database writes performed: ${receipt.databaseWritesPerformed}`,
    "",
    "FINDINGS"
  ];
  for (const finding of receipt.findings) lines.push(`- ${finding.passed ? "PASS" : "FAIL"} ${finding.code}: ${finding.message}`);
  lines.push("", "MANUAL EVIDENCE", `- Installation observed: ${receipt.manualEvidence.installationObserved}`, `- Offline reload observed: ${receipt.manualEvidence.offlineReloadObserved}`);
  return lines.join("\n");
}

File: infrastructure/deployment/qualification/formatting.py
import json
from .contracts import DeploymentQualification

def format_human(result: DeploymentQualification) -> str:
    lines=["I006 DEPLOYMENT PACKAGE QUALIFICATION", "="*72, f"Status: {result.status}", f"Database writes performed: {result.database_writes_performed}", "", "FINDINGS"]
    lines.extend(f"- {'PASS' if f.passed else 'FAIL'} {f.code}: {f.message}" for f in result.findings)
    return "\n".join(lines)

def format_json(result: DeploymentQualification) -> str:
    return json.dumps(result.to_dict(), indent=2, sort_keys=True)
